local/bin/py/build/actions/pull_and_push_file.py

```diff
@@ -30,13 +31,17 @@ def pull_and_push_file(content, content_dir):
             # remove h1 if exists
             file_content = re.sub(re.compile(r"^#{1}(?!#)(.*)", re.MULTILINE), "", file_content, count=1)
             file_content = TEMPLATE.format(front_matter=front_matter, content=file_content.strip())
-    with open(
-        "{}{}{}".format(
-            content_dir,
-            content["options"]["dest_path"][1:],
-            basename(content["options"]["file_name"]),
-        ),
-        mode="w+",
-        encoding="utf-8",
-    ) as f:
-        f.write(file_content)
+        elif "datadog-agent" in content["repo_name"] and "config_template.yaml" in "".join(content["globs"]):
+            process_agent_config(file_content)
```

Comment:
I was originally wondering if we should make this a separate script instead of adding this logic into push_pull but i see why as its very similar behavior. We can make bigger architecture plans to improve this kind of thing instead.

Reply:
Also just noticed in the config we have a repo called `puppet-datadog-agent`  is that intended to run this logic?

Reply:
@davidejones no `puppet-datadog-agent` isn't intended to run this logic, i noticed that as well so to address i've added the second condition there to make sure `config_template.yaml` is present in the globs.